### PR TITLE
make RobertaForMaskedLM implementation identical to fairseq

### LINKED ISSUE #1874: Disparitry with Fairseq Roberta implementation for predicting the mask token
## ❓ Questions & Help
When experimenting with the fill mask functionality on the fairseq repo I realized there is a disparity with the results I get from huggingface implementation. Wondering if there is a mismatch between the model released here and their latest release.

Thanks

### PR DESCRIPTION
**Pull request by BramVanroy**
closes https://github.com/huggingface/transformers/issues/1874

The implementation of RoBERTa in `transformers` differs from the original implementation in [fairseq](https://github.com/pytorch/fairseq/tree/master/fairseq/models/roberta), as results showed (cf. https://github.com/huggingface/transformers/issues/1874). I have documented my findings here https://github.com/huggingface/transformers/issues/1874#issuecomment-588359143 and made the corresponding changes accordingly in this PR.

Someone should check, however, that removing `get_output_embeddings()` does not have any adverse side-effects.

### CODE REVIEW

**Comment by sshleifer** [comment]
Awesome. Could you check if the existing @slow tests break for Roberta, and add a new one that hardcodes the fairseq logits from your example and makes sure we also return them. Trying to avoid accidental breakage. Thanks again!

**Comment by BramVanroy** [comment]
@sshleifer Not sure how I would hardcode a tensor of size 1, 12, 50265. Can I just add a small pickled file to the test instead?

**Comment by BramVanroy** [comment]
Sorry for the close, had to do some rebasing.
